src/stronghold/mod.rs

```diff
@@ -135,24 +132,37 @@ impl StrongholdAdapterBuilder {
     ///
     /// [`password()`]: Self::password()
     /// [`timeout()`]: Self::timeout()
-    pub fn try_build(mut self) -> Result<StrongholdAdapter> {
+    pub fn try_build(mut self, snapshot_path: PathBuf) -> Result<StrongholdAdapter> {
         // In any case, Stronghold - as a necessary component - needs to be present at this point.
         let stronghold = if let Some(stronghold) = self.stronghold {
             stronghold
         } else {
-            let system = ActorSystem::new()?;
-            let client_path = PRIVATE_DATA_CLIENT_PATH.to_vec();
-            let options = Vec::new();
-
-            Arc::new(Mutex::new(Stronghold::init_stronghold_system(
-                system,
-                client_path,
-                options,
-            )))
+            Stronghold::default()
         };
 
+        if let Some(key) = &self.key {
+            // TODO I don't think this.
+            let key_provider = KeyProvider::try_from((**key).clone())?;
+            let result = stronghold.load_client_from_snapshot(
+                PRIVATE_DATA_CLIENT_PATH,
+                &key_provider,
+                &SnapshotPath::from_path(&snapshot_path),
+            );
+            if let Err(iota_stronghold::ClientError::Inner(err_msg)) = result {
```

Comment:
The `ClientError::Inner` variant is a compromise to support other kind of errors inside the `ClientError` enum, but we can discuss that, if there is a better solution.